frontend/shared/src/types/input-props.d.ts

```diff
@@ -0,0 +1,12 @@
+import { RegisterOptions, UseFormRegister } from 'react-hook-form';
+
+type InputProps<T, V = string> = {
+  id: NonNullable<Parameters<UseFormRegister<T>>[0]>;
+  registerOptions?: RegisterOptions<T>;
+  initialValue?: V;
+  onChange?: (value: V) => void;
+  label?: string | React.ReactNode;
```

Comment:
I think `ReactNode` include `string`

```suggestion
  label?: React.ReactNode;
```

Reply:
fixed